mne/connectivity/cfc.py

```diff
@@ -0,0 +1,497 @@
+# Authors: Praveen Sripad <praveen.sripad@rwth-aachen.de>
+#          Alexandre Gramfort <alexandre.gramfort@telecom-paristech.fr>
+#
+# License: BSD (3-clause)
+
+import numpy as np
+from numpy.random import shuffle
+from scipy import stats
+from scipy.signal import hilbert, gaussian, hanning
+
+from ..stats import fdr_correction
+from ..time_frequency import morlet
+from ..time_frequency.tfr import cwt
+from ..preprocessing.peak_finder import peak_finder
+from ..filter import band_pass_filter
+from ..parallel import parallel_func
+from ..utils import logger, check_random_state
+
+
+def _abs_cwt(x, W):
+    """ Aux function """
+    return np.abs(cwt(x, [W], use_fft=False).ravel()).astype(np.float)
+
+
+def make_surrogate_data(data):
+    """ Returns a copy of the shuffled surrogate data.
+
+    Parameters
+    ----------
+    data : ndarray, shape (n_epochs, n_times)
+        Data to be shuffled.
+
+    Returns
+    -------
+    shuffled_data : ndarray, shape (n_epochs, n_times)
+        The shuffled data, with each epoch shuffled across n_times.
+    """
+    shuffled_data = data.copy()
+    map(shuffle, shuffled_data)
+    return shuffled_data
```

Comment:
actually look to be a lot of good surrogate functions in there

On Thu, Jul 16, 2015 at 7:15 PM, Chris Holdgraf <choldgraf@gmail.com> wrote:

> I also found this:
>
>
> http://www.pik-potsdam.de/~donges/pyunicorn/_modules/pyunicorn/timeseries/surrogates.html#Surrogates.correlated_noise_surrogates
>
> On Thu, Jul 16, 2015 at 7:09 PM, Denis A. Engemann <
> notifications@github.com> wrote:
>
>> In mne/connectivity/cfc.py
>> <https://github.com/mne-tools/mne-python/pull/1388#discussion_r34812123>:
>>
>> > +def make_surrogate_data(data):
>> > +    """ Returns a copy of the shuffled surrogate data.
>> > +
>> > +    Parameters
>> > +    ----------
>> > +    data : ndarray, shape (n_epochs, n_times)
>> > +        Data to be shuffled.
>> > +
>> > +    Returns
>> > +    -------
>> > +    shuffled_data : ndarray, shape (n_epochs, n_times)
>> > +        The shuffled data, with each epoch shuffled across n_times.
>> > +    """
>> > +    shuffled_data = data.copy()
>> > +    map(shuffle, shuffled_data)
>> > +    return shuffled_data
>>
>> I have it locally, can contribute it.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/mne-tools/mne-python/pull/1388/files#r34812123>.
>>
>
>